File: shoppingm8_fe/lib/list/product/dto/productRequestDto.dart
import '../productCategory.dart';

class ProductRequestDto {
  final String name;
  final num requiredAmount;
  final String unit;
  final ProductCategory category;

  ProductRequestDto({this.name, this.requiredAmount, this.unit, this.category});

  Map toJson() {
    Map map = Map();
    map['name'] = name;
    map['requiredAmount'] = requiredAmount;
    map['unit'] = unit;
    map['category'] = category.toString().replaceFirst("ProductCategory.", "");
    return map;
  }
}

File: ShoppingM8-be/src/main/java/edu/pw/shoppingm8/product/api/dto/ProductRequestDto.java
package edu.pw.shoppingm8.product.api.dto;

import javax.validation.constraints.NotNull;

import lombok.Value;

@Value
public class ProductRequestDto {
    @NotNull
    Long listId;
    @NotNull
    String name;
    @NotNull
    String unit;
    @NotNull
    String category;
    @NotNull
    Double requiredAmount;

}